blockly-core/core/ui/block_space/block_space.js

```diff
@@ -819,11 +858,66 @@ Blockly.BlockSpace.prototype.getScrollableSize = function(metrics) {
 };
 
 /**
- * Given desired new scrollX and scrollY positions, scroll to position,
- * clamping to within allowable scroll boundaries.
- * @param {number} newScrollX new target pan-right (+) offset
- * @param {number} newScrollY new target pan-down (+) offset
+ * Returns a box representing the size of the underlying editable canvas
+ * @returns {goog.math.Box}
+ */
+Blockly.BlockSpace.prototype.getScrollableBox = function() {
+  var scrollableSize = this.getScrollableSize(this.getMetrics());
+  return new goog.math.Box(0, scrollableSize.width, scrollableSize.height, 0);
+};
+
+/**
+ * Returns a box representing the position of the viewport in the coordinate
+ * space of the underlying canvas.
+ * @returns {goog.math.Box}
  */
+Blockly.BlockSpace.prototype.getViewportBox = function() {
+  var metrics = this.getMetrics();
+  return new goog.math.Box(
+    this.getScrollOffsetY(),
+    this.getScrollOffsetX() + metrics.viewWidth,
+    this.getScrollOffsetY() + metrics.viewHeight,
+    this.getScrollOffsetX());
+};
+
+Blockly.BlockSpace.prototype.panIfOverEdge = function (block, mouseX, mouseY) {
+  this.scrollOnBlockDragHandler_.panIfOverEdge(block, mouseX, mouseY);
+};
+
+Blockly.BlockSpace.prototype.stopAutoScrolling = function () {
+  this.scrollOnBlockDragHandler_.stopAutoScrolling();
+};
+
+/**
+ * Given a block, scrolls the viewport to contain the block (plus a small
+ * margin).
+ * @param {Blockly.Block} block
+ */
+Blockly.BlockSpace.prototype.scrollIntoView = function (block) {
+  var blockBox = block.getBox();
+  var currentView = this.getViewportBox();
+
+  var boxOverflows = Blockly.getBoxOverflow(currentView, blockBox);
+  Blockly.addToNonZeroSides(boxOverflows,
+    Blockly.BlockSpace.DROPPED_BLOCK_PAN_MARGIN);
+  this.scrollToDelta(boxOverflows.right - boxOverflows.left,
+    boxOverflows.bottom - boxOverflows.top);
+};
+
+Blockly.BlockSpace.prototype.scrollDeltaWithAnySelectedBlock = function (scrollDx, scrollDy,
```

Comment:
Needs doc - I'm not sure from the name what the mouseX and mouseY parameters do.